development/build/transforms/remove-fenced-code.js

```diff
@@ -0,0 +1,416 @@
+const path = require('path');
+const { PassThrough, Transform } = require('stream');
+const { BuildTypes } = require('../utils');
+
+module.exports = {
+  createRemoveFencedCodeTransform,
+  removeFencedCode,
+};
+
+class RemoveFencedCodeTransform extends Transform {
+  /**
+   * A transform stream that calls {@link removeFencedCode} on the complete
+   * string contents of the file read by Browserify.
+   *
+   * @param {string} filePath - The path to the file being transformed.
+   * @param {string} buildType - The type of the current build process.env.
+   */
+  constructor(filePath, buildType) {
+    super();
+    this.filePath = filePath;
+    this.buildType = buildType;
+    this._fileBuffers = [];
+  }
+
+  // This function is called whenever data is written to the stream.
+  // It concatenates all buffers for the current file into a single buffer.
+  _transform(buffer, _encoding, next) {
+    this._fileBuffers.push(buffer);
+    next();
+  }
+
+  // "flush" is called when all data has been written to the
+  // stream, immediately before the "end" event is emitted.
+  // It applies the transform to the concatenated file contents.
+  _flush(end) {
+    const [fileContent] = removeFencedCode(
+      this.filePath,
+      this.buildType,
+      Buffer.concat(this._fileBuffers).toString('utf8'),
+    );
+
+    this.push(fileContent);
+    end();
+  }
+}
+
+/**
+ * A factory for a Browserify transform that removes fenced code from all
+ * JavaScript source files. The transform is applied to files with the following
+ * extensions:
+ *   - `.js`
+ *   - `.cjs`
+ *   - `.mjs`
+ *
+ * For details on how the transform mutates source files, see
+ * {@link removeFencedCode} and the documentation.
+ *
+ * @param {string} buildType - The type of the current build.
+ * @returns {(filePath: string) => Transform} The transform function.
+ */
+function createRemoveFencedCodeTransform(buildType) {
+  if (!(buildType in BuildTypes)) {
```

Comment:
in keyword on objects includes the prototype chain eg `'toString' in {} // true`